test/remote/recovery_code_tests.js

```diff
@@ -138,6 +139,41 @@ describe('remote recovery codes', function () {
     })
   })
 
+  describe('should notify user when recovery codes are low', () => {
+    beforeEach(() => {
+      // Create a new unverified session to test recovery codes
+      return Client.login(config.publicUrl, email, password)
+        .then((response) => {
+          client = response
+          return client.emailStatus()
+        })
+        .then((res) => assert.equal(res.sessionVerified, false, 'session not verified'))
+    })
+
+    it('should consume recovery code and verify session', () => {
+      return client.consumeRecoveryCode(recoveryCodes[0], {metricsContext})
+        .then((res) => {
+          assert.equal(res.remaining, 7, 'correct remaining codes')
+          return server.mailbox.waitForEmail(email)
+        })
+        .then((emailData) => {
+          assert.equal(emailData.headers['x-template-name'], 'postConsumeRecoveryCodeEmail', 'correct template sent')
+          return client.consumeRecoveryCode(recoveryCodes[1], {metricsContext})
+        })
+        .then((res) => {
+          assert.equal(res.remaining, 6, 'correct remaining codes')
+          return server.mailbox.waitForEmail(email)
+        })
+        .then((emails) => {
+          // The order in which the emails are sent is not guaranteed, test for both possible templates
+          const email1 = emails[0].headers['x-template-name']
+          const email2 = emails[1].headers['x-template-name']
+          assert.equal(true, email1 === 'postConsumeRecoveryCodeEmail' || email1 === 'lowRecoveryCodesEmail', 'correct template sent')
```

Comment:
nit: This is super pedantic, this test does not assert that two emails are different.